### ISSUE TITLE
Make PSA Crypto work with Cipher light

### ISSUE THREAD

**Issue by mpg** [Mbed-TLS/mbedtls#8063]
- Change it from "depends on `CIPHER_C`" to "auto-enables `CIPHER_LIGHT` in `build-info.h`".
- Re-enable it in `all.sh` component `test_cipher_light_only`
- Re-enable everything that was disabled only because it depended on this: `USE_PSA_CRYPTO`, TLS, etc.

Note: might need to adjust some declared dependencies for this, for example TLS should depend on `CIPHER_C || USE_PSA_CRYPTO`.

~Note: once this task is done, as well as #8059, #8060 and #8061, ideally `test_cipher_light_only` will have only `CIPHER_C` disabled an nothing else.~ Edit: it will have more disabled in accordance [to the current goals](https://github.com/Mbed-TLS/mbedtls/issues/8337): 

**Depends on:** #8058

**Comment by mpg** [comment]
On hold pending strategy discussion.

**Comment by mpg** [comment]
Closing - no longer inline with the current strategy, and something similar has been achieved already.
